### PR TITLE
[WEB-10519] - Update JS SDK to allow for Assets endpoint to be accessed with basic auth

### PR DESCRIPTION
**Pull request by luke-wyeth**
All changes made **only** to the DA service get Assets endpoint
* When we are in build mode (no access token is available) add the CDN auth secret so that the DA service can be properly accessed
* If an access token is present, use bearer token. If no access token, return use basic auth header

### CODE REVIEW

File: src/DigitalAssets.js [hunk 1]
```diff
@@ -106,19 +106,32 @@ export default class DigitalAssetsService extends Service {
...
+    }
+
+    if (!this._sws.accessToken) {
+      headers['x-serato-cdn-auth'] = this.xSeratoCdnAuthHeader()
+    }
+
+    // If an access token is present, use bearer token. If no access token, return use basic auth header
     return this.fetch(
-      this.bearerTokenAuthHeader(),
+      this._sws.accessToken ? this.bearerTokenAuthHeader() : this.basicAuthHeader(),
```
Comment (ben-farrelly) [inline src/DigitalAssets.js:120]: Maybe it's better to specifically check for `''` here?
Something like `'false'` or `'0'` would pass this.... but also, to be fair, they're not valid tokens.

**Review by ben-farrelly — APPROVED** [review]
Looks good to me, nice work guys

**Review by sameeraroshan-serato — APPROVED** [review]

File: src/DigitalAssets.js (hunk 1 above)
Comment (luke-wyeth) [reply src/DigitalAssets.js:120]: @sameeraroshan-serato what do you reckon about this?

Comment (sameeraroshan-serato) [reply src/DigitalAssets.js:120]: As Ben said those are not valid tokens. Even though it tries to fetch basicAuthHeader in that case, basicAuthHeader is available only from server side. So I don't think that would be an issue.

**Pull request merged**